packages/core/utils/src/dml/__tests__/entity-builder.spec.ts

```diff
@@ -1554,162 +1553,25 @@ describe("Entity builder", () => {
   })
 
   describe("Entity builder | primaryKey", () => {
-    test("should create both id fields and primaryKey fields", () => {
+    test("should infer primaryKeys from a model", () => {
       const user = model.define("user", {
-        id: model.id(),
+        id: model.id().primaryKey(),
         email: model.text().primaryKey(),
-        account_id: model.number().primaryKey(),
+        account_id: model.number(),
       })
 
       const entityBuilder = createMikrORMEntity()
       const User = entityBuilder(user)
-
-      expectTypeOf(new User()).toMatchTypeOf<{
-        id: string
-        email: string
-        account_id: number
-      }>()
-
       const metaData = MetadataStorage.getMetadataFromDecorator(User)
 
-      expect(metaData.properties).toEqual({
-        id: {
-          reference: "scalar",
-          type: "string",
-          columnType: "text",
-          name: "id",
-          nullable: false,
-          getter: false,
-          setter: false,
-        },
-        email: {
-          columnType: "text",
-          name: "email",
-          nullable: false,
-          primary: true,
-          reference: "scalar",
-          type: "string",
-        },
-        account_id: {
-          columnType: "integer",
-          name: "account_id",
-          nullable: false,
-          primary: true,
-          reference: "scalar",
```

Comment:
I have to delete a lot of code from this test, because we are re-creating/registering the same entity with Mikro ORM (via "User" as the model name).

As a result, Mikro ORM is performing all sorts of meta-data merging under the hood and making it super hard to reason about the assertions. 

So either we break this test into multiple tests, or we should create unique entities in this test.